packages/core/src/components/graphs/bar.ts

```diff
@@ -1,22 +1,30 @@
 // Internal Imports
 import { Component } from "../component";
+import { BarOrientationOptions } from "../../interfaces";
 
 export class Bar extends Component {
 	// Gets the correct width for bars based on options & configurations
 	protected getBarWidth() {
-		const mainXAxis = this.services.axes.getMainXAxis();
 		const options = this.model.getOptions();
+		let mainAxis;
 
-		if (!mainXAxis.scale.step) {
+		// if its horizontal we want to use the Y axis to determine instead of X
+		if (options.orientation === BarOrientationOptions.HORIZONTAL) {
```

Comment:
This seems like the last place `optoins.orientation` is used ... From what I can see we should be able to replace it with a call to `this.services.cartesianScales.getOrientation`